### ISSUE TITLE
Call to a member function getDelay() on null

### ISSUE THREAD

**Issue by mostafaznv** [Intervention/gif#8]
I encountered this error when trying to read a specific GIF file

**Code:**
```php
$imageManager = ImageManager::gd();
$imageManager->read('path/to/file.gif');
```

**Error:**
```log
Call to a member function getDelay() on null at intervention/gif/src/Splitter.php:131
```

I expect this code to work and read my GIF file, as it works with other files (the same code with a different GIF file works perfectly).

----

**Note:**
If I change the driver to `imagick`, everything works fine. It seems to be a problem related to the `GD` driver/library, but it doesn't happen with all GIF files.

----

**Environment:**
 - PHP Version: 8.3.0
 - OS: MacOS
 - Intervention Image Version: 3.1.0
 - GD: bundled (2.1.0 compatible) (GIF Read Support: enabled)
 
<br />

 **Gif File:**
![gif-2](https://github.com/Intervention/gif/assets/7619687/6fd8aac2-5896-48a7-be4e-66dcb6571020)

**Comment by mostafaznv** [comment]
I understand that my example code is based on the syntax from `intervention/image`. However, since I believe the error is caused by `intervention/gif`, I have created the issue here.

**Comment by olivervogel** [comment]
Thanks for reporting. I will look into this.

**Comment by olivervogel** [comment]
The error lies in the decoding of the individual DataBlocks. These can normally be either a GraphicBlock or a SpecialPurposeBlock.

The library assumes that a DataBlock is either a GraphicBlock or a SpecialPurposeBlock and expects the following schema for a DataBlock, which is also described in the GIF specification:

- GraphicBlock ([GraphicControlExtension] | ((ImageDescriptor [LocalColorTable]ImageData) | PlainTextExtension))
- SpecialPurposeBlock (ApplicationExtension | CommentExtension)

According to the specification, a GraphicBlock **always** begins with a GraphicControlExtension. However, in the file you provided I have seen that the GraphicControlExtension is directly followed by the ApplicationExtension, which actually does not correspond to the GIF specification. But apparently it does result in a functioning file.

**Comment by mostafaznv** [comment]
Thank you for the clarification.
Personally, I'm fine with skipping this file, as I tested importing it into an online Photoshop tool and re-exported it and it worked fine.
However, I raised this issue to explore potential solutions, aiming to use the original file without any modifications. While I'm uncertain about the underlying reasons, it appears Imagick can handle it somehow.

Nevertheless, if you believe this isn't a bug, I'm open to closing the issue.
Additionally, considering the likelihood that other users might encounter similar files, there might be a way to address this—whether through a potential fix for the issue or by implementing some form of validation (although I'm unsure of the feasibility).

**Comment by olivervogel** [comment]
I think the best way is to modify the library so that it can handle GIF files that are not 100% compliant with the specification. My research has shown that this is not always the case in alot of files.

**Comment by mostafaznv** [comment]
Should I open an issue on their [repository](https://github.com/libgd/libgd)? I'm concerned that I may not be able to articulate it properly. I'm not very familiar with either the GIF specification or the potential reasons behind this issue in their codebase.

However, if you believe it's necessary to raise this on their repository, I can certainly do so.

**Comment by olivervogel** [comment]
> Should I open an issue on their [repository](https://github.com/libgd/libgd)?

No, not necessary. The issue is already in the right place.

**Comment by olivervogel** [comment]
Will be available in intervention/image 3.2.2 tomorrow.

**Comment by mostafaznv** [comment]
Thank you very much 🔥
